### PR TITLE
Upgrade Slate and implement new schema

### PR DESCRIPTION
**Pull request by Zhouzi**
This PR upgrades Slate to the latest version and updates the schema to use the new definition.

### CODE REVIEW

File: lib/validation/validateNode.js [hunk 1]
```diff
@@ -1,245 +1,35 @@
...
- * A rule that ensure code lines only contain one text
- * node.
- */
-function onlyText(opts: Options) {
-    return {
-        match: node => node.type === opts.lineType,
-
-        validate(node) {
-            const { nodes } = node;
+        if (!opts.allowMarks && node.type === opts.lineType) {
```
Comment (Soreine) [inline lib/validation/validateNode.js]: Can't you use the marks validation from the new schema ?

https://docs.slatejs.org/other-packages/slate-schema-violations#node_mark_invalid

**Comment by Zhouzi** [comment]
@Soreine thanks, I thought it was not possible, it's now fixed :) I updated Slate's documentation to mention that option (it was missing): https://github.com/ianstormtaylor/slate/pull/1717

File: lib/validation/schema.js [hunk 2]
```diff
@@ -41,6 +41,12 @@ function schema(opts: Options): Object {
             }
         }
     };
+
+    if (!opts.allowMarks) {
+        baseSchema.blocks[opts.lineType].marks = [];
```
Comment (Soreine) [inline lib/validation/schema.js]: Much more concise <3

**Comment by Soreine** [comment]
You rock ❤️

**Pull request merged**